File: cameraview/src/main/java/com/base/cameraview/preview/FilterCameraPreview.java
package com.base.cameraview.preview;


import androidx.annotation.NonNull;

import com.base.cameraview.filter.Filter;


/**
 * A preview that support GL filters defined through the {@link Filter} interface.
 * <p>
 * The preview has the responsibility of calling {@link Filter#setSize(int, int)}
 * whenever the preview size changes and as soon as the filter is applied.
 */
public interface FilterCameraPreview {

    /**
     * Sets a new filter.
     *
     * @param filter new filter
     */
    void setFilter(@NonNull Filter filter);

    /**
     * Returns the currently used filter.
     *
     * @return currently used filter
     */
    @SuppressWarnings("unused")
    @NonNull
    Filter getCurrentFilter();
}

File: cameraview/src/main/java/com/base/cameraview/preview/FilterCameraPreview.kt
package com.base.cameraview.preview

import com.base.cameraview.filter.Filter


/**
 * A preview that support GL filters defined through the [Filter] interface.
 *
 *
 * The preview has the responsibility of calling [Filter.setSize]
 * whenever the preview size changes and as soon as the filter is applied.
 */
interface FilterCameraPreview {
    /**
     * Sets a new filter.
     *
     * @param filter new filter
     */
    fun setFilter(filter: Filter)

    val currentFilter: Filter
}
